### ISSUE TITLE
🐛 Bug Report: OpenSearch password visible in devtools config screen

### ISSUE THREAD

**Issue by bforbis** [backstage/backstage#23300]
### 📜 Description

When using opensearch as a search provider, I am seeing the `search.elasticsearch.auth.password` config visible in plaintext in the devtools.

This configuration is already annotated with the `@visibility secret` within the [search-backend-module-elasticsearch](https://github.com/backstage/backstage/blob/master/plugins/search-backend-module-elasticsearch/config.d.ts#L205) plugin.

See similar issue: https://github.com/backstage/backstage/issues/18812

### 👍 Expected behavior

Passwords should be masked by the devtools plugin when annotated as a secret

### 👎 Actual Behavior with Screenshots

**app-config.yaml**
```yaml
search:
  elasticsearch:
    provider: opensearch
    node: http://search:9200
    auth:
      username: admin
      # Don't worry, this isn't a real password!
      password: 2xZ%NFnbMDk9WNoV
```

![image](https://github.com/backstage/backstage/assets/16994570/ef521130-7957-43ef-80b1-d1e14dcc861e)

### 👟 Reproduction steps

1. Follow setup instructions for [search-backend-module-elasticsearch](https://github.com/backstage/backstage/tree/master/plugins/search-backend-module-elasticsearch#search-backend-module-elasticsearch)
2. Follow setup instructions for [devtools](https://github.com/backstage/backstage/tree/master/plugins/devtools)
3. Navigate to http://localhost:3000/devtools/config

### 🖥️ Your Environment

```
OS:   Darwin 23.3.0 - darwin/arm64
node: v18.18.0
yarn: 1.22.19
cli:  0.25.2 (installed)
backstage:  1.23.3

Dependencies:
  @backstage/app-defaults                                          1.5.0
  @backstage/backend-app-api                                       0.5.13
  @backstage/backend-common                                        0.20.2, 0.21.2
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.5
  @backstage/backend-plugin-api                                    0.6.12
  @backstage/backend-tasks                                         0.5.17
  @backstage/catalog-client                                        1.6.0
  @backstage/catalog-model                                         1.4.4
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.3
  @backstage/cli                                                   0.25.2
  @backstage/config-loader                                         1.6.2
  @backstage/config                                                1.1.1
  @backstage/core-app-api                                          1.12.0
  @backstage/core-compat-api                                       0.2.0
  @backstage/core-components                                       0.12.5, 0.13.10, 0.14.0
  @backstage/core-plugin-api                                       1.9.0
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.3
  @backstage/eslint-plugin                                         0.1.5
  @backstage/frontend-plugin-api                                   0.6.0
  @backstage/integration-aws-node                                  0.1.9
  @backstage/integration-react                                     1.1.24
  @backstage/integration                                           1.9.0
  @backstage/plugin-adr-backend                                    0.4.9
  @backstage/plugin-adr-common                                     0.2.20
  @backstage/plugin-adr                                            0.6.13
  @backstage/plugin-analytics-module-ga4                           0.2.0
  @backstage/plugin-api-docs                                       0.11.0
  @backstage/plugin-app-backend                                    0.3.60
  @backstage/plugin-app-node                                       0.1.12
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.4
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.3
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.7
  @backstage/plugin-auth-backend-module-github-provider            0.1.9
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.9
  @backstage/plugin-auth-backend-module-google-provider            0.1.9
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.7
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.9
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.5
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.2
  @backstage/plugin-auth-backend-module-okta-provider              0.0.5
  @backstage/plugin-auth-backend                                   0.21.2
  @backstage/plugin-auth-node                                      0.4.7
  @backstage/plugin-bazaar-backend                                 0.3.10
  @backstage/plugin-bazaar                                         0.2.22
  @backstage/plugin-catalog-backend-module-aws                     0.3.6
  @backstage/plugin-catalog-backend-module-bitbucket-server        0.1.25
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.9
  @backstage/plugin-catalog-backend                                1.17.2
  @backstage/plugin-catalog-common                                 1.0.21
  @backstage/plugin-catalog-graph                                  0.4.0
  @backstage/plugin-catalog-import                                 0.10.6
  @backstage/plugin-catalog-node                                   1.7.2
  @backstage/plugin-catalog-react                                  1.10.0
  @backstage/plugin-catalog                                        1.17.0
  @backstage/plugin-devtools-backend                               0.2.9
  @backstage/plugin-devtools-common                                0.1.8
  @backstage/plugin-devtools                                       0.1.9
  @backstage/plugin-entity-validation                              0.1.15
  @backstage/plugin-events-backend                                 0.2.21
  @backstage/plugin-events-node                                    0.2.21
  @backstage/plugin-github-actions                                 0.6.11
  @backstage/plugin-kubernetes-common                              0.7.4
  @backstage/plugin-linguist-backend                               0.5.9
  @backstage/plugin-linguist-common                                0.1.2
  @backstage/plugin-linguist                                       0.1.15
  @backstage/plugin-org                                            0.6.20
  @backstage/plugin-permission-backend                             0.5.35
  @backstage/plugin-permission-common                              0.7.12
  @backstage/plugin-permission-node                                0.7.23
  @backstage/plugin-permission-react                               0.4.20
  @backstage/plugin-proxy-backend                                  0.4.10
  @backstage/plugin-scaffolder-backend-module-azure                0.1.4
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.2
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.2
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.2
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.4
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.2
  @backstage/plugin-scaffolder-backend-module-github               0.2.2
  @backstage/plugin-scaffolder-backend-module-gitlab               0.2.15
  @backstage/plugin-scaffolder-backend                             1.21.2
  @backstage/plugin-scaffolder-common                              1.5.0
  @backstage/plugin-scaffolder-node                                0.3.2
  @backstage/plugin-scaffolder-react                               1.8.0
  @backstage/plugin-scaffolder                                     1.18.0
  @backstage/plugin-search-backend-module-catalog                  0.1.16
  @backstage/plugin-search-backend-module-elasticsearch            1.3.15
  @backstage/plugin-search-backend-module-pg                       0.5.21
  @backstage/plugin-search-backend-module-techdocs                 0.1.16
  @backstage/plugin-search-backend-node                            1.2.16
  @backstage/plugin-search-backend                                 1.5.2
  @backstage/plugin-search-common                                  1.2.10
  @backstage/plugin-search-react                                   1.7.6
  @backstage/plugin-search                                         1.4.6
  @backstage/plugin-tech-radar                                     0.6.13
  @backstage/plugin-techdocs-backend                               1.9.5
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.5
  @backstage/plugin-techdocs-node                                  1.11.4
  @backstage/plugin-techdocs-react                                 1.1.16
  @backstage/plugin-techdocs                                       1.10.0
  @backstage/plugin-user-settings                                  0.8.1
  @backstage/release-manifests                                     0.0.11
  @backstage/repo-tools                                            0.6.2
  @backstage/test-utils                                            1.5.0
  @backstage/theme                                                 0.2.19, 0.5.1
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by bforbis** [comment]
@awanlin, FYI since you seem to be mostly maintaining this plugin

**Comment by aramissennyeydd** [comment]
@Rugvip @davidfestal I think this line is causing some issues, https://github.com/backstage/backstage/commit/2fb527dd34dab57917005d6e0b51a1c5332d4f99#diff-c3638c66544f0291784a8973946226e439203759979696c291a9d3fd57ccd7ecR45. Seems like a no-op filter? AFAICT this means that no secrets are being detected/redacted for anything that's using `createConfigSecretEnumerator`.

Even with the above, the core issue here is https://github.com/backstage/backstage/issues/21430, the config schema and used value don't line up. That will need to be fixed for this to be fixed.

**Comment by freben** [comment]
Made some PRs (linked above) to try to address this.

**Comment by awanlin** [comment]
Thanks everyone for the quick turn around on this! 🚀

**Comment by davidfestal** [comment]
> @Rugvip @davidfestal I think this line is causing some issues, [2fb527d#diff-c3638c66544f0291784a8973946226e439203759979696c291a9d3fd57ccd7ecR45](https://github.com/backstage/backstage/commit/2fb527dd34dab57917005d6e0b51a1c5332d4f99#diff-c3638c66544f0291784a8973946226e439203759979696c291a9d3fd57ccd7ecR45). Seems like a no-op filter? AFAICT this means that no secrets are being detected/redacted for anything that's using `createConfigSecretEnumerator`.
> 

Oh, That's a pity.  This filter was added temporarily during tests of the PR, and I should have removed it before committing. Sorry for this.

**Comment by davidfestal** [comment]
@freben @awanlin Would this fix be added in a minor 1.23.x release ?
